misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/DiagramUtil.java

```diff
@@ -53,6 +55,23 @@ public static JsonElement getSyntaxTreeJSON(Document srcFile, SemanticModel sema
         return syntaxTreeJson;
     }
 
+    public static JsonElement getSyntaxTreeJSONByRange(Node node, SemanticModel semanticModel) {
+        JsonElement syntaxTreeJson;
+        try {
+            SyntaxTreeMapGenerator mapGenerator = new SyntaxTreeMapGenerator(semanticModel);
+
+            if (node.kind() == SyntaxKind.LIST) {
+                syntaxTreeJson = mapGenerator.transformSyntaxNode(node.parent());
+            } else {
+                syntaxTreeJson = mapGenerator.transformSyntaxNode(node);
+            }
+        } catch (NullPointerException e) {
```

Comment:
Why do we capture the NPE here? Can't we handle this by having a null check? Also, it would be better to catch `Throwable e` in order to handle the exceptions gracefully at the server, without throwing the exception